biz.aQute.repository/test/aQute/bnd/repository/osgi/OSGiRepositoryTest.java

```diff
@@ -47,66 +49,80 @@ protected void setUp() throws Exception {
 
 	public void testSimple() throws Exception {
 		try (OSGiRepository r = new OSGiRepository();) {
-			Map<String, String> map = new HashMap<>();
-			map.put("locations", fnx.getBaseURI("/repo/minir5.xml")
-				.toString());
-			map.put("cache", cache.getPath());
-			map.put("max.stale", "10000");
-			r.setProperties(map);
-			Processor p = new Processor();
-			HttpClient httpClient = new HttpClient();
-			httpClient.setCache(cache);
-			httpClient.setRegistry(p);
-			p.addBasicPlugin(httpClient);
-			p.setBase(ws);
-			p.addBasicPlugin(Workspace.createStandaloneWorkspace(p, ws.toURI()));
-			r.setRegistry(p);
-
-			final AtomicInteger tasks = new AtomicInteger();
-
-			p.addBasicPlugin(new ProgressPlugin() {
-
-				@Override
-				public Task startTask(final String name, int size) {
-					System.out.println("Starting " + name);
-					tasks.incrementAndGet();
-					return new Task() {
-
-						@Override
-						public void worked(int units) {
-							System.out.println("Worked " + name + " " + units);
-						}
-
-						@Override
-						public void done(String message, Throwable e) {
-							System.out.println("Done " + name + " " + message);
-						}
-
-						@Override
-						public boolean isCanceled() {
-							return false;
-						}
-					};
-				}
-			});
+			assertTrue(testRepo(r).check());
```

Comment:
`testRepo` returns a `Processor` that should be closed.

```java
try (Processor p = testRepo(r)) {
  assertTrue(p.check());
}
```

Reply:
this is a test case and free processor that will be gc'ed